doc/src/lexer_tutorial/002_raw_delimited_content.md

```diff
@@ -0,0 +1,346 @@
+You could resolve this with a custom lexer (treated in the next section).
+
+But the simplest thing to do in this scenario is to read the string delimiters and the string content as a *single token*, like so:
+
+```lalrpop
+pub Var: Var = <r"[a-z]"> => <>.chars().next().unwrap().into();
+
+pub Lit: Lit = <l:r#""[a-z ]*""#> => l[1..l.len()-1].into();
+
+pub Stmt: Stmt = <Var> "=" <Lit> => (<>).into();
+```
+
+(Note that this form of the grammar does not require any `match` statement;
+there is no longer any ambiguity between the different regular expressions that
+drive the tokenizer.)
+
+With this definition of the grammar, all of these tests pass:
+
+```
+#[test]
+fn homerun() {
+    assert_eq!(nobol5::VarParser::new().parse("x"), Ok('x'.into()));
+    assert_eq!(nobol5::LitParser::new().parse(r#""abc""#), Ok("abc".into()));
+    assert_eq!(nobol5::StmtParser::new().parse(r#"x = "a""#), Ok(('x', "a").into()));
+    assert_eq!(nobol5::StmtParser::new().parse(r#"y = "bc""#), Ok(('y', "bc").into()));
+    assert_eq!(nobol5::StmtParser::new().parse(r#"z = "xyz""#), Ok(('z', "xyz").into()));
+    assert_eq!(nobol5::StmtParser::new().parse(r#"z = "x""#), Ok(('z', "x").into()));
+    assert_eq!(nobol5::StmtParser::new().parse(r#"z = "x y z""#), Ok(('z', "x y z").into()));
+}
+```
+
+Furthermore, we can now remove other artificial limits in our language.
+For example, we can make our identifiers more than one character:
+
+```lalrpop
+pub Var: Var = <r"[a-z]+"> => <>.into()
+```
+
+which, with suitable changes to the library code, works out fine.
+
+#### Escape sequences
+
+Our current string literals are allowed to hold a small subset of the full space of characters.
+
+If we wanted to generalize it to be able to hold arbitrary characters, we would
+need some way to denote the delimiter character `"` in ths string content.
+
+The usual way to do this is via an escape sequence: `\"`, which is understood by
+the lexical analyzer as *not* ending the string content.
+
+We can generalize the regular expression in our new `Lit` rule to handle this:
+
+```lalrpop
+pub Lit: Lit = <l:r#""(\\\\|\\"|[^"])*""#> => l[1..l.len()-1].into();
+```
+
+However, depending on your data model, this is not quite right. In particular:
+the string is produced still has the escaping backslashes embedded in it.
+
+
+As a concrete example, with the above definition for `Lit`, this test:
+
+```rust
+#[test]
+fn popfly() {
+    assert_eq!(nobol6::StmtParser::new().parse(r#"z = "\"\\""#), Ok(('z', "\"\\").into()));
+}
+```
+
+yields this output:
+
+```
+thread 'popfly' panicked at 'assertion failed: `(left == right)`
+  left: `Ok(Stmt(Var('z'), Lit("\\\"\\\\")))`,
+ right: `Ok(Stmt(Var('z'), Lit("\"\\")))`', doc/nobol/src/main.rs:91:5
+```
+
+This can be readily addressed by adding some code to post-process the token to remove the
+backslashes:
+
+```lalrpop
+pub Lit: Lit = <l:r#""(\\\\|\\"|[^"])*""#> =>
+    l[1..l.len()-1].replace("\\\"", "\"").replace("\\\\", "\\").into();
+```
```

Comment:
````suggestion
```

where `apply_string_escapes` is a helper routine that searches for backslashes in the content and performs the corresponding replacement with the character denoted by the escape sequence.
````